### PR TITLE
Only load locales in I18n.available_locales

### PR DESCRIPTION
**Pull request by keichan34**
Pulls a lot from [devise-i18n](https://github.com/tigrish/devise-i18n). Also uses a Railtie to load i18n files instead of the engine (I was having some trouble with the Engine loading the files at all with Rails 4.2.x)

### CODE REVIEW

File: doorkeeper-i18n.gemspec [hunk 1]
```diff
@@ -17,4 +17,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency "rake"
   s.add_development_dependency "rspec"
   s.add_development_dependency "i18n-spec", "~> 0.6.0"
+  s.add_development_dependency "railties", ">= 0"
```
Comment (tute) [inline doorkeeper-i18n.gemspec]: We don't need to add this dependency here. `doorkeeper-i18n` depends on `doorkeeper`, and that project depends on `railties`.

**Comment by tute** [comment]
Looks great, thank you! A minor comment, then it's ready for merge I believe.

**Comment by keichan34** [comment]
Looks like `doorkeeper` [wasn't listed in the dependencies](https://github.com/doorkeeper-gem/doorkeeper-i18n/blob/9a951e2f625b8188dc7be7eb5a5032d7858675f0/doorkeeper-i18n.gemspec), so I added it in that commit.

**Comment by tute** [comment]
Thank you!

**Pull request closed without merging**